compiler/resolution/functionResolution.cpp

```diff
@@ -13776,44 +13779,170 @@ void checkDuplicateDecorators(Type* decorator, Type* decorated, Expr* ctx) {
   }
 }
 
-void checkSurprisingGenericDecls(DefExpr* def, bool isField) {
-  if (def == nullptr) {
+static bool isBuiltinGenericType(Type* t) {
+  return isBuiltinGenericClassType(t) ||
+         t == dtAnyComplex || t == dtAnyImag || t == dtAnyReal ||
+         t == dtAnyBool || t == dtAnyEnumerated ||
+         t == dtNumeric || t == dtIntegral ||
+         t == dtIteratorRecord || t == dtIteratorClass ||
+         t == dtAnyPOD ||
+         t == dtOwned || t == dtShared ||
+         t == dtAnyRecord;
```

Comment:
Please double-check whether the above needs dtBorrowed, dtUnmanaged, dtAnyManagementAnyNilable, dtAnyNilable(??).

Reply:
These are covered by `isBuiltinGenericClassType` which is called as the first option in the `||` clauses.